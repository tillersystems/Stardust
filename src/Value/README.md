# Value

### Usage

```jsx
import { Value } from '@tillersystems/stardust';
```

<!-- STORY -->

### Properties

- `children` - anything that can be rendered: numbers, strings, elements or an array (or fragment).
- `className` - Class needed by styled component.

| propName    | propType | defaultValue | isRequired |
| ----------- | :------: | :----------: | :--------: |
| `children`  |  `node`  |    `null`    |     -      |
| `className` | `string` |     `''`     |     -      |

### Example

```jsx
import { Value } from '@tillersystems/stardust';

render() {
  return (
    <Value positive>+10</Value>
  );
}
```
