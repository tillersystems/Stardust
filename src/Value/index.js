import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

/**
 * A value is a number that can be positive, negative or neutral (0).
 *
 * @return {jsx}
 */

/* eslint-disable no-unused-vars */

const Value = ({ children, className, positive, negative }) => (
  <div className={className}>{children}</div>
);

/**
 * PropTypes Validation
 */
const { bool, node, string } = PropTypes;
Value.propTypes = {
  /** Anything that can be rendered: numbers, strings, elements or an array (or fragment) */
  children: node,

  /** className needed by styled-components */
  className: string,

  /** If the value is positive  */
  positive: bool,

  /** If the value is negative */
  negative: bool,
};

/**
 * Default props
 */
Value.defaultProps = {
  children: null,
  className: '',
  positive: false,
  negative: false,
};

export default styled(Value)`
  color: ${({ theme: { palette } }) => palette.darkBlue};

  ${({ negative }) =>
    negative &&
    css`
      color: ${({ theme: { palette } }) => palette.failure.default};
    `};

  ${({ positive }) =>
    positive &&
    css`
      color: ${({ theme: { palette } }) => palette.success.default};
    `};
`;
