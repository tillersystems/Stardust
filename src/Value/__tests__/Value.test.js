import React from 'react';
import Value from '..';

describe('<Value />', () => {
  test('should render withouth a problem', () => {
    const { container } = render(<Value>0</Value>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render as negative', () => {
    const { container } = render(<Value negative>-10</Value>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render as postive', () => {
    const { container } = render(<Value positive>+10</Value>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
