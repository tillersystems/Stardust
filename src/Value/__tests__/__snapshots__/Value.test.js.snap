// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Value /> should render as negative 1`] = `
.c0 {
  color: hsl(213,17%,20%);
  color: hsl(6,79%,65%);
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "1.6rem",
          "bigInt": 1.6,
          "default": "1.2rem",
          "defaultInt": 1.2,
          "h1": "3.6rem",
          "h1Int": 3.6,
          "h2": "3rem",
          "h2Int": 3,
          "h3": "2.6rem",
          "h3Int": 2.6,
          "h4": "2.2rem",
          "h4Int": 2.2,
          "h5": "2.0rem",
          "h5Int": 2,
          "h6": "1.8rem",
          "h6Int": 1.8,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.4rem",
          "mediumInt": 1.4,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "avatar": Object {
          "alizarin": "hsl(6, 78%, 57%)",
          "carrot": "hsl(28, 80%, 52%)",
          "emerald": "hsl(145, 63%, 49%)",
          "river": "hsl(204, 70%, 53%)",
          "turquoise": "hsl(168, 76%, 42%)",
          "wisteria": "hsl(282, 44%, 47%)",
        },
        "black": "hsl(0, 0%, 0%)",
        "bodyBackground": "hsl(220, 27%, 94%)",
        "darkBlue": "hsl(213, 17%, 20%)",
        "darkGrey": "hsl(206,16%,60%)",
        "failure": Object {
          "dark": "#e74432",
          "darker": "#e42f1b",
          "default": "hsl(6, 79%, 65%)",
        },
        "googleBrandRed": "hsl(5, 81%, 56%)",
        "lightGrey": "hsl(207,22%,90%)",
        "mediumGrey": "hsl(206,23%,69%)",
        "mysticGrey": "hsl(216,45%,98%)",
        "primary": Object {
          "dark": "#1873a0",
          "darker": "#14638a",
          "default": "hsl(200, 74%, 46%)",
        },
        "secondary": Object {
          "default": "hsl(0, 100%, 100%)",
        },
        "spaceGrey": "hsl(207,13%,45%)",
        "success": Object {
          "dark": "#6da333",
          "darker": "#60902d",
          "default": "hsl(89, 53%, 52%)",
        },
        "warning": Object {
          "dark": "#f59a24",
          "darker": "#f48f0b",
          "default": "hsl(34, 91%, 65%)",
        },
        "white": "hsl(0, 100%, 100%)",
        "whiteOpacity": [Function],
      },
    }
  }
>
  <Styled(Value)
    negative={true}
  >
    <Value
      className="c0"
      negative={true}
      positive={false}
    >
      <div
        className="c0"
      >
        -10
      </div>
    </Value>
  </Styled(Value)>
</ThemeProvider>
`;

exports[`<Value /> should render as postive 1`] = `
.c0 {
  color: hsl(213,17%,20%);
  color: hsl(89,53%,52%);
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "1.6rem",
          "bigInt": 1.6,
          "default": "1.2rem",
          "defaultInt": 1.2,
          "h1": "3.6rem",
          "h1Int": 3.6,
          "h2": "3rem",
          "h2Int": 3,
          "h3": "2.6rem",
          "h3Int": 2.6,
          "h4": "2.2rem",
          "h4Int": 2.2,
          "h5": "2.0rem",
          "h5Int": 2,
          "h6": "1.8rem",
          "h6Int": 1.8,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.4rem",
          "mediumInt": 1.4,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "avatar": Object {
          "alizarin": "hsl(6, 78%, 57%)",
          "carrot": "hsl(28, 80%, 52%)",
          "emerald": "hsl(145, 63%, 49%)",
          "river": "hsl(204, 70%, 53%)",
          "turquoise": "hsl(168, 76%, 42%)",
          "wisteria": "hsl(282, 44%, 47%)",
        },
        "black": "hsl(0, 0%, 0%)",
        "bodyBackground": "hsl(220, 27%, 94%)",
        "darkBlue": "hsl(213, 17%, 20%)",
        "darkGrey": "hsl(206,16%,60%)",
        "failure": Object {
          "dark": "#e74432",
          "darker": "#e42f1b",
          "default": "hsl(6, 79%, 65%)",
        },
        "googleBrandRed": "hsl(5, 81%, 56%)",
        "lightGrey": "hsl(207,22%,90%)",
        "mediumGrey": "hsl(206,23%,69%)",
        "mysticGrey": "hsl(216,45%,98%)",
        "primary": Object {
          "dark": "#1873a0",
          "darker": "#14638a",
          "default": "hsl(200, 74%, 46%)",
        },
        "secondary": Object {
          "default": "hsl(0, 100%, 100%)",
        },
        "spaceGrey": "hsl(207,13%,45%)",
        "success": Object {
          "dark": "#6da333",
          "darker": "#60902d",
          "default": "hsl(89, 53%, 52%)",
        },
        "warning": Object {
          "dark": "#f59a24",
          "darker": "#f48f0b",
          "default": "hsl(34, 91%, 65%)",
        },
        "white": "hsl(0, 100%, 100%)",
        "whiteOpacity": [Function],
      },
    }
  }
>
  <Styled(Value)
    positive={true}
  >
    <Value
      className="c0"
      negative={false}
      positive={true}
    >
      <div
        className="c0"
      >
        +10
      </div>
    </Value>
  </Styled(Value)>
</ThemeProvider>
`;

exports[`<Value /> should render withouth a problem 1`] = `
.c0 {
  color: hsl(213,17%,20%);
}

<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1200,
        "xs": 0,
      },
      "dimensions": Object {
        "big": "3.1rem",
        "bigInt": 3.1,
        "huge": "4.5rem",
        "hugeInt": 4.5,
        "medium": "1.5rem",
        "mediumInt": 1.5,
        "radius": "0.4rem",
        "radiusInt": 0.4,
        "small": "0.7rem",
        "smallInt": 0.7,
        "tiny": "0.3rem",
        "tinyInt": 0.3,
      },
      "fonts": Object {
        "family": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
        "size": Object {
          "big": "1.6rem",
          "bigInt": 1.6,
          "default": "1.2rem",
          "defaultInt": 1.2,
          "h1": "3.6rem",
          "h1Int": 3.6,
          "h2": "3rem",
          "h2Int": 3,
          "h3": "2.6rem",
          "h3Int": 2.6,
          "h4": "2.2rem",
          "h4Int": 2.2,
          "h5": "2.0rem",
          "h5Int": 2,
          "h6": "1.8rem",
          "h6Int": 1.8,
          "huge": "6.4rem",
          "hugeInt": 6.4,
          "medium": "1.4rem",
          "mediumInt": 1.4,
          "small": "0.7rem",
          "smallInt": 0.7,
          "tiny": "0.3rem",
          "tinyInt": 0.3,
        },
        "spacing": Object {
          "big": "1.5px",
          "medium": "1px",
          "small": "0.5px",
        },
        "weight": Object {
          "normal": 400,
          "thick": 600,
          "thin": 200,
        },
      },
      "palette": Object {
        "avatar": Object {
          "alizarin": "hsl(6, 78%, 57%)",
          "carrot": "hsl(28, 80%, 52%)",
          "emerald": "hsl(145, 63%, 49%)",
          "river": "hsl(204, 70%, 53%)",
          "turquoise": "hsl(168, 76%, 42%)",
          "wisteria": "hsl(282, 44%, 47%)",
        },
        "black": "hsl(0, 0%, 0%)",
        "bodyBackground": "hsl(220, 27%, 94%)",
        "darkBlue": "hsl(213, 17%, 20%)",
        "darkGrey": "hsl(206,16%,60%)",
        "failure": Object {
          "dark": "#e74432",
          "darker": "#e42f1b",
          "default": "hsl(6, 79%, 65%)",
        },
        "googleBrandRed": "hsl(5, 81%, 56%)",
        "lightGrey": "hsl(207,22%,90%)",
        "mediumGrey": "hsl(206,23%,69%)",
        "mysticGrey": "hsl(216,45%,98%)",
        "primary": Object {
          "dark": "#1873a0",
          "darker": "#14638a",
          "default": "hsl(200, 74%, 46%)",
        },
        "secondary": Object {
          "default": "hsl(0, 100%, 100%)",
        },
        "spaceGrey": "hsl(207,13%,45%)",
        "success": Object {
          "dark": "#6da333",
          "darker": "#60902d",
          "default": "hsl(89, 53%, 52%)",
        },
        "warning": Object {
          "dark": "#f59a24",
          "darker": "#f48f0b",
          "default": "hsl(34, 91%, 65%)",
        },
        "white": "hsl(0, 100%, 100%)",
        "whiteOpacity": [Function],
      },
    }
  }
>
  <Styled(Value)>
    <Value
      className="c0"
      negative={false}
      positive={false}
    >
      <div
        className="c0"
      >
        0
      </div>
    </Value>
  </Styled(Value)>
</ThemeProvider>
`;
